### PR TITLE
docs: translate PureComponent page

### LINKED ISSUE #553: `PureComponent`
Page: https://id.react.dev/reference/react/PureComponent

### PR DESCRIPTION
**Pull request by rhenaldkarrel**
Closes #553 

## Description

Translate the PureComponent page.
Page URL: https://id.react.dev/reference/react/PureComponent

### CODE REVIEW

File: src/content/reference/react/PureComponent.md [hunk 1]
```diff
@@ -26,11 +26,10 @@ class Greeting extends PureComponent {
 
 ---
 
-## Reference {/*reference*/}
+## Referensi {/*reference*/}
 
 ### `PureComponent` {/*purecomponent*/}
-
-To skip re-rendering a class component for same props and state, extend `PureComponent` instead of [`Component`:](/reference/react/Component)
+Untuk melewatkan render ulang sebuah komponen kelas untuk props dan state yang sama, perluas `PureComponent`, bukan [`Component`:](/reference/react/Component)
```
Comment (mazipan) [inline src/content/reference/react/PureComponent.md:32]: Missing new line after the heading

Comment (mazipan) [reply src/content/reference/react/PureComponent.md:32]: extend in this refer to `extends` in class component.
I suggest to keep as it is, using italic font.
Or we can introduce for the first time by adding the original word.

E.g: `memperluas (*extend*)`

File: src/content/reference/react/PureComponent.md [hunk 2]
```diff
@@ -4,13 +4,13 @@ title: PureComponent
...
 
-We recommend defining components as functions instead of classes. [See how to migrate.](#alternatives)
+Kami menyarankan untuk membuat komponen sebagai fungsi, bukan kelas. [Lihat cara migrasinya.](#alternatives)
 
 </Pitfall>
 
 <Intro>
 
-`PureComponent` is similar to [`Component`](/reference/react/Component) but it skips re-renders for same props and state. Class components are still supported by React, but we don't recommend using them in new code.
+`PureComponent` mirip dengan [`Component`](/reference/react/Component) tetapi melewatkan proses render ulang untuk props dan state yang sama. Komponen kelas masih didukung oleh React, tetapi kami tidak menyarankan untuk menggunakannya dalam kode baru.
```
Comment (mazipan) [inline src/content/reference/react/PureComponent.md:13]: render = `*render*` --> italic

File: src/content/reference/react/PureComponent.md (hunk 1 above)
Comment (mazipan) [inline src/content/reference/react/PureComponent.md:32]: render = `*render*` --> italic

File: src/content/reference/react/PureComponent.md [hunk 3]
```diff
@@ -203,6 +202,6 @@ label {
 
 <Note>
 
-Unlike `PureComponent`, [`memo`](/reference/react/memo) does not compare the new and the old state. In function components, calling the [`set` function](/reference/react/useState#setstate) with the same state [already prevents re-renders by default,](/reference/react/memo#updating-a-memoized-component-using-state) even without `memo`.
+Tidak seperti `PureComponent`, [`memo`](/reference/react/memo) tidak membandingkan state yang baru dan state yang lama. Pada komponen fungsi, memanggil [fungsi `set`](/reference/react/useState#setstate) dengan state yang sama [sudah mencegah render ulang secara default,](/reference/react/memo#updating-a-memoized-component-using-state) bahkan tanpa `memo`.
```
Comment (mazipan) [inline src/content/reference/react/PureComponent.md:205]: render = `*render*` --> italic

File: src/content/reference/react/PureComponent.md (hunk 1 above)
Comment (rhenaldkarrel) [reply src/content/reference/react/PureComponent.md:32]: I think using `*extends*` is more relevant somehow

```suggestion
Untuk melewatkan render ulang sebuah komponen kelas untuk props dan state yang sama, *extends* `PureComponent`, bukan [`Component`:](/reference/react/Component)
```

**Review by mazipan — APPROVED** [review]

**Review by resir014 — APPROVED** [review]

**Pull request merged**